packages/webview_flutter/webview_flutter_wkwebview/test/src/web_kit_webview_widget_test.dart

```diff
@@ -205,6 +259,79 @@ void main() {
       });
 
       group('$WebSettings', () {
+        testWidgets('javascriptMode', (WidgetTester tester) async {
+          await buildWidget(
+            tester,
+            creationParams: CreationParams(
+              webSettings: WebSettings(
+                userAgent: const WebSetting<String?>.absent(),
+                javascriptMode: JavascriptMode.unrestricted,
+                hasNavigationDelegate: false,
+              ),
+            ),
+          );
+
+          verify(mockPreferences.setJavaScriptEnabled(true));
+        });
+
+        testWidgets('hasNavigationDelegate', (WidgetTester tester) async {
+          await buildWidget(
+            tester,
+            creationParams: CreationParams(
+              webSettings: WebSettings(
+                userAgent: const WebSetting<String?>.absent(),
+                hasNavigationDelegate: true,
+              ),
+            ),
+          );
+
+          verify(mockNavigationDelegate
+              .setDecidePolicyForNavigationAction(argThat(isNotNull)));
+        });
+
+        testWidgets('userAgent', (WidgetTester tester) async {
+          await buildWidget(
+            tester,
+            creationParams: CreationParams(
+              webSettings: WebSettings(
+                userAgent: const WebSetting<String?>.of('myUserAgent'),
+                hasNavigationDelegate: false,
+              ),
+            ),
+          );
+
+          verify(mockWebView.setCustomUserAgent('myUserAgent'));
+        });
+
+        testWidgets('zoomEnabled', (WidgetTester tester) async {
+          await buildWidget(
+            tester,
+            creationParams: CreationParams(
+              webSettings: WebSettings(
+                userAgent: const WebSetting<String?>.absent(),
+                zoomEnabled: false,
+                hasNavigationDelegate: false,
+              ),
+            ),
+          );
+
+          final WKUserScript zoomScript =
+              verify(mockUserContentController.addUserScript(captureAny))
+                  .captured
+                  .first as WKUserScript;
+          expect(zoomScript.isMainFrameOnly, isTrue);
+          expect(zoomScript.injectionTime,
+              WKUserScriptInjectionTime.atDocumentEnd);
+          expect(
+            zoomScript.source,
+            "var meta = document.createElement('meta');"
+            "meta.name = 'viewport';"
+            "meta.content = 'width=device-width, initial-scale=1.0, maximum-scale=1.0,"
+            "user-scalable=no';"
+            "var head = document.getElementsByTagName('head')[0];head.appendChild(meta);",
+          );
+        });
+
```

Comment:
There should be a test that enabling zoom doesn't destroy the other scripts.

Reply:
done